docs/tutorials/node-app.rst

```diff
@@ -0,0 +1,144 @@
+Bundle a Node.js app into a ROCK
+********************************
+
+This tutorial describes the steps needed to bundle a typical Node.js application
+into a ROCK.
+
+Prerequisites
+-------------
+- snap enabled system (https://snapcraft.io/docs/installing-snapd)
+- LXD installed
+  (https://documentation.ubuntu.com/lxd/en/latest/howto/initialize/)
+- Docker installed (https://snapcraft.io/docker)
+- a text editor
+
+
+Install Rockcraft
+-----------------
+
+Install Rockcraft on your host:
+
+.. literalinclude:: code/node-app/task.yaml
+    :language: bash
+    :start-after: [docs:install-rockcraft]
+    :end-before: [docs:install-rockcraft-end]
+    :dedent: 2
+
+
+Project Setup
+-------------
+
+Starting in an empty folder, create a ``src/`` subdirectory. Inside it, add two
+files:
+
+The first one is the ``package.json`` listing of dependencies, with the
+following contents:
+
+.. literalinclude:: code/node-app/src/package.json
+    :caption: package.json
+    :language: json
+
+The second file is our sample app, a simple "hello world" server. Still inside
+``src/``, add the following contents to ``server.js``:
+
+.. literalinclude:: code/node-app/src/server.js
+    :caption: server.js
+    :language: javascript
+
+Next, we'll setup the Rockcraft project. In the original empty folder, create
+an empty file called ``rockcraft.yaml``. Then add the following snippets, one
+after the other:
+
+Add the metadata that describes your ROCK, such as its name and licence:
+
+.. literalinclude:: code/node-app/rockcraft.yaml
+    :caption: rockcraft.yaml
+    :language: yaml
+    :start-after: [docs:rock-metadata]
+    :end-before: [docs:rock-metadata-end]
+
+Add the container entrypoint, as a `Pebble`_ service:
+
+.. literalinclude:: code/node-app/rockcraft.yaml
+    :caption: rockcraft.yaml
+    :language: yaml
+    :start-after: [docs:pebble-service]
+    :end-before: [docs:pebble-service-end]
+
+Finally, add a part that describes how to build the app created in the ``src/``
+directory using the ``npm`` plugin:
+
+.. literalinclude:: code/node-app/rockcraft.yaml
+    :caption: rockcraft.yaml
+    :language: yaml
+    :start-after: [docs:app-part]
+    :end-before: [docs:app-part-end]
+
+
+Pack the ROCK with Rockcraft
+----------------------------
+
+To build the ROCK, run:
+
+.. literalinclude:: code/node-app/task.yaml
+    :language: bash
+    :start-after: [docs:build-rock]
+    :end-before: [docs:build-rock-end]
+    :dedent: 2
+
+At the end of the process, a file named ``my-node-app_0.1_amd64.rock`` should be
```

Comment:
Could you actually test the OCI archive name? Since we have a history of changing the naming conventions for the output artefact, it would be nice to test this string in order to be future-proof and preemptively detect any coming changes to the output filename.

Reply:
It's a good idea, I definitely agree with the sentiment. I updated the docs to do a `ls my-node-app_1.0_amd64.rock`, which looks slightly awkward in the tutorial but will break if we change this name.